implementations/rust/daemon/src/cli.rs

```diff
@@ -0,0 +1,366 @@
+    )]
+    role: ChannelRole,
+
+    /// Define the channel responder address, currently obtained from a responder node.
+    #[structopt(
+        long,
+        required_if("role", "initiator"),
+        required_if("role", "init"),
+        help = r#"Address used to reach channel "responder" on remote machine"#
+    )]
+    channel_responder_address: Option<String>,
+
+    /// Define the worker address, currently obtained from a responder node.
+    #[structopt(
+        long,
+        required_if("role", "initiator"),
+        required_if("role", "init"),
+        help = r#"Address used to reach "worker" on remote machine"#
+    )]
+    worker_address: Option<String>,
+
+    /// Define which private key to use as the initiator's identity.
+    #[structopt(
+        long,
+        required_if("role", "initiator"),
+        required_if("role", "init"),
+        help = "Name of the private key to use for the identity of the channel initiator"
+    )]
+    identity_name: Option<String>,
+
+    /// Define the public key needed to communicate with the channel responder.
+    #[structopt(
+        long,
+        required_if("role", "initiator"),
+        required_if("role", "init"),
+        help = "The public key provided by channel responder"
+    )]
+    responder_public_key: Option<String>,
+
+    // TODO: expose `control` and `control_port` once runtime configuration is needed.
+    #[structopt(
+        short,
+        long,
+        help = "Execute `ockamd` in control-mode, otherwise will start as a long-running process",
+        set = Hidden,
+    )]
+    control: bool,
+    #[structopt(
+        short = "p",
+        long = "port",
+        default_value = "11199",
+        help = "Port for runtime configuration updates",
+        set = Hidden,
+    )]
+    control_port: u16,
+}
+
+impl Default for Args {
+    fn default() -> Self {
+        Self {
+            control: false,
+            control_port: DEFAULT_CONFIG_PORT,
+            input: InputKind::Stdin,
+            output: OutputKind::Stdout,
+            local_socket: SocketAddr::from_str(DEFAULT_LOCAL_SOCKET)
+                .expect("bad default set for local socket"),
+            decrypt_output: true,
+            vault: VaultKind::Filesystem,
+            vault_path: PathBuf::from("ockamd_vault"),
+            role: ChannelRole::Responder,
+            channel_responder_address: None,
+            worker_address: None,
+            identity_name: None,
+            responder_public_key: None,
+        }
+    }
+}
+
+impl Args {
+    /// Parse the command line options into the Args struct.
+    pub fn parse() -> Args {
+        let mut args = Args::from_args();
+
+        // validate provided arguments & override possibly fallible options
+        match args.output_kind() {
+            OutputKind::Channel(_) => {
+                // disallow output to be decrypted if it's to be sent over a secure channel
+                if args.decrypt_output {
```

Comment:
Mostly curious about why this is the case.

Reply:
Good callout 😆 a couple reason: In a demo, it is useful to be able to show the data in both states, and as a requirement for the near-term deliverable we need to just write cleartext messages to `stdout` on the responder end.